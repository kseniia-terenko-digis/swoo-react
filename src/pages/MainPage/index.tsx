import Button from '../../shared/components/Button';
import Container from '../../shared/components/Container';
import Counter from '../../shared/components/Counter';
import { useCounter } from '../../shared/hooks/useCounter';

const MainPage: React.FC = () => {
  const { counterValue, decrementCounter, incrementCounter } = useCounter();

  return (
    <Container
      bg="blue"
      height="100%"
      display="flex"
      flexDirection="column"
      alignItems="center"
      justifyContent="center"
    >
      <Button onClick={incrementCounter}>+ 1</Button>
      <Container my={104}>
        <Counter>{counterValue}</Counter>
      </Container>
      <Button onClick={decrementCounter}>- 1</Button>
    </Container>
  );
};

export default MainPage;
