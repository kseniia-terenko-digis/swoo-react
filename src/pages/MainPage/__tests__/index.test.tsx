import { shallow } from 'enzyme';
import MainPage from '..';

const mockHookResult = {
  counterValue: 10,
  decrementCounter: () => {},
  incrementCounter: () => {},
};

jest.mock('../../../shared/hooks/useCounter', () => ({
  useCounter: () => mockHookResult,
}));

describe('MainPage component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<MainPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
