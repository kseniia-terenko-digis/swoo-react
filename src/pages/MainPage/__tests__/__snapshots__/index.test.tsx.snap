// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component matches snapshot 1`] = `
<Memo(styled.div)
  alignItems="center"
  bg="blue"
  display="flex"
  flexDirection="column"
  height="100%"
  justifyContent="center"
>
  <Memo(styled.button)
    onClick={[Function]}
  >
    + 1
  </Memo(styled.button)>
  <Memo(styled.div)
    my={104}
  >
    <Memo(styled.span)>
      10
    </Memo(styled.span)>
  </Memo(styled.div)>
  <Memo(styled.button)
    onClick={[Function]}
  >
    - 1
  </Memo(styled.button)>
</Memo(styled.div)>
`;
